recipes/quickfix/all/conanfile.py

```diff
@@ -1,92 +1,110 @@
-    description = "QuickFIX is a free and open source implementation of the FIX protocol"
-    topics = ("conan", "QuickFIX", "FIX", "Financial Information Exchange", "libraries", "cpp")
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"fPIC": [True, False],
-               "shared": [True, False],
-               "with_ssl":  [True, False],
-               "with_postgres": [True, False],
-               "with_mysql": [None, "libmysqlclient"]}
-    default_options = {"fPIC": True,
-                       "shared": False,
-                       "with_ssl": False,
-                       "with_postgres": False,
-                       "with_mysql": None}
-    generators = "cmake"
-    exports_sources = "patches/**"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def requirements(self):
-        if self.options.with_ssl:
-            self.requires("openssl/1.1.1q")
-
-        if self.options.with_postgres:
-            self.requires("libpq/14.2")
-
-        if self.options.with_mysql == "libmysqlclient":
-            self.requires("libmysqlclient/8.0.29")
+    topics = ("FIX", "Financial Information Exchange", "libraries", "cpp")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [True, False],
+        "with_postgres": [True, False],
+        "with_mysql": [None, "libmysqlclient"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": False,
+        "with_postgres": False,
+        "with_mysql": None,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_ssl:
+            self.requires("openssl/[>=1.1 <4]")
+
+        if self.options.with_postgres:
+            self.requires("libpq/15.3")
+
+        if self.options.with_mysql == "libmysqlclient":
+            self.requires("libmysqlclient/8.0.31")
 
     def validate(self):
         if self.settings.os == "Windows" and self.options.shared:
             raise ConanInvalidConfiguration("QuickFIX cannot be built as shared lib on Windows")
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            raise ConanInvalidConfiguration("QuickFIX doesn't support ARM compilation")  # See issue: https://github.com/quickfix/quickfix/issues/206
+        if is_apple_os(self) and self.settings.arch == "armv8":
```

Comment:
Fundamentally, it doesn't really make sense to replace Macos by is_apple_os in this specific case. It was just a way to pass CI when macOS M1 builds have beenn introduced. Actually it's broken for arm in general.